### PR TITLE
fix: remove aws-sdk V2; import aws-sdk V3 packages (#637)

### PR DESCRIPTION
**Pull request by david-mcafee**
## fix (Apollo V3): remove aws-sdk V2; import aws-sdk V3 packages (#637)

*Issue #, if available:*
  - #637 

*Description of changes:*
 - fix(aws-appsync-auth-link): update auth-link and signer
 - Removing `aws-sdk` v2, and importing only the v3 packages we need. Includes minor type updates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/aws-appsync-auth-link/src/auth-link.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ type KeysWithType<O, T> = {
 type AuthOptionsNone = { type: AUTH_TYPE.NONE };
 type AuthOptionsIAM = {
     type: KeysWithType<typeof AUTH_TYPE, AUTH_TYPE.AWS_IAM>,
-    credentials: (() => Credentials | CredentialsOptions | Promise<Credentials | CredentialsOptions | null>) | Credentials | CredentialsOptions | null,
+    credentials: (() => Credentials | Provider<Credentials> | Promise<Credentials | Provider<Credentials> | null>) | Credentials | Provider<Credentials> | null,
```
Comment (elorzafe) [inline packages/aws-appsync-auth-link/src/auth-link.ts:117]: `Provider<Credentials>` and `CredentialsOptions` are the same?

File: packages/aws-appsync-auth-link/src/signer/signer.ts [hunk 2]
```diff
@@ -7,16 +7,23 @@ See the License for the specific language governing permissions and limitations
 */
 global.Buffer = global.Buffer || require('buffer').Buffer; // Required for aws sigv4 signing
 
-var url = require('url'),
-    crypto = require('aws-sdk/global').util.crypto;
+var url = require('url');
 
-var encrypt = function (key, src, encoding = '') {
-    return crypto.lib.createHmac('sha256', key).update(src, 'utf8').digest(encoding);
+var { Sha256 } = require('@aws-crypto/sha256-universal')
```
Comment (elorzafe) [inline packages/aws-appsync-auth-link/src/signer/signer.ts:12]: Can we try to use https://github.com/aws-amplify/amplify-js/blob/main/packages/core/src/Signer.ts#L15 this approach instead, that hash is sync that will make this a non-breaking change. What do you think?

**Review by elorzafe — COMMENTED** [review]
This is great @david-mcafee thank you!

Can you check if this approach is more appropriate so we can avoid doing a breaking change release?

**Review by elorzafe — COMMENTED** [review]
This looks very good!

I would only check if you couid use `@aws-crypto/sha256-js`(12KB) its around half the size of `@aws-crypto/sha256-universal` (19KB) and it would not increment the bundle size when using with Amplify. I would test that on node to see if works I would go with that one.

**Review by elorzafe — APPROVED** [review]
Thanks @david-mcafee 🏅

**Comment by david-mcafee** [comment]
# AWS-SDK upgrade (Apollo V3)

## Manual Testing Checklist:

[ X ] node-apollo-V3-API-Key

[ X ] node-apollo-V3-Cognito

[ X ] node-apollo-V3-IAM

[ X ] react-apollo-V3-API-Key

[ X ] react-apollo-V3-Cognito

[ X ] react-apollo-V3-IAM

[ X ] react-native-apollo-V3-API-Key

[ X ] react-native-apollo-V3-Cognito

[ X ] react-native-apollo-V3-IAM

**Pull request merged**